content/arc-sqlmi-chaos-mesh/index.mdx

```diff
@@ -0,0 +1,248 @@
+
+So recently, I was tasked with trying my absolute best _to break_ SQL Server Availability Group by any means possible. The definition of broken is:
+
+<Callout>
+
+"_Do whatever you want to the system (besides deleting `mdf/ldf` files etc., AKA no irrecoverable damage), for any duration of time, with the end goal of leaving the system in a state where it cannot self-heal, and requires manual intervention (e.g. intelligently crafted T-SQL commands that must be manually run by some human being)._
+
+_If it requires manual intervention, the system is broken, and that's a major bug._"
+
+</Callout>
+
+This time, I had Kubernetes on my side, with the SQL engine running in disposable Linux containers ([Azure Arc-enabled SQL Managed Instance, Business-Critical Tier](https://learn.microsoft.com/en-us/azure/azure-arc/data/service-tiers#service-tier-comparison)), and the latest-and-greatest Chaos Engineering software from CNCF - called [Chaos Mesh](https://www.cncf.io/projects/chaosmesh/) - offering many novel [Custom Resource Defintions](https://github.com/chaos-mesh/chaos-mesh#chaos-operator) that can be used to declare layers upon layers of controlled, systematic degradations (rather than one-off imperative scripts with a single entry and exit point like, restarting a VM, then moving on to cause some other mischief in a linear fashion, etc.).
+
+Having [pondered about questions like this](https://www.reddit.com/r/kubernetes/comments/tfga4y/eli5_why_is_it_not_advisable_to_run_databases_in/) about the scary unknowns of hosting data inside ephemeral containers (K8s Pods), I was feeling pretty good about my chances of discovering a failure-mode that triumphed `sqlservr.exe`(s).
+
+## High Availability Architecture and possible faliure points
+
+<Callout>
+
+The key to try and make a system miserably fail is to first understand it's architecture and key lifelines. That way, the kinds of failures we can produce _really_ try to hurt the system by cutting off one or more of those lifelines.
+
+</Callout>
+
+### HA Kubernetes Operator and Pendulum Analogy
+
+- Azure Arc-enabled SQL Managed Instance Business Critical tier, besides wrapping all SQL replicas behind an Availability Group Listener, comes with a dedicated [Kubernetes Operator](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) called the `arc-ha-orchestrator`.
+
+  - An Operator is a dedicated piece of computer program responsible for monitoring the health state of some system it "operates", based on some system specific [Finite-state machine](https://en.wikipedia.org/wiki/Finite-state_machine) definition (basically, business logic).
+
+    It then uses the observed signals to figure out the Current-State, and perform some small, pre-defined, intelligent actions to "nudge" (rather than force), the system towards a healthy (i.e. Desired) state. That system could be anything, SQL Server, a pendulum, a self-driving car etc.
```

Comment:
""nudge" (rather than force), the system"
You shouldn't have a comma here

Reply:
"anything, SQL Server" change that comma to a colon to start a list:

"anything: SQL Server"

Reply:
Done and done